independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/SseUtil.java

```diff
@@ -138,7 +139,7 @@ private static String serialiseDataToString(ResteasyReactiveRequestContext conte
             if (writer.isWriteable(entityClass, entityType, Serialisers.NO_ANNOTATION, mediaType)) {
                 // FIXME: spec doesn't really say what headers we should use here
                 writer.writeTo(entity, entityClass, entityType, Serialisers.NO_ANNOTATION, mediaType,
-                        Serialisers.EMPTY_MULTI_MAP, baos);
+                        new QuarkusMultivaluedHashMap<>(), baos);
```

Comment:
Instead of passing empty map as headers, we should pass a copy of currently set headers.

In that way, an implementation of `MessageBodyWriter` could detect it should serialize the object for Server Sent Events, so it makes no sense to set headers.

Also, it makes sense to warn people that those headers will be dropped. For example, to check if the headers been changed after calling write, we can log a WARN message.

Reply:
If you can provide a test showing why this is important, we can certainly reconsider

Reply:
See https://github.com/quarkusio/quarkus/issues/31587 and https://github.com/cloudevents/sdk-java/issues/533 for more context